test/api/v3/integration/groups/POST-groups_groupId_leave.js

```diff
@@ -0,0 +1,167 @@
+import {
+  generateUser,
+  generateChallenge,
+  checkExistence,
+  sleep,
+} from '../../../../helpers/api-v3-integration.helper';
+
+describe('POST /groups/:groupId/leave', () => {
+  let user;
+  let userToInvite;
+
+  beforeEach(async () => {
+    user = await generateUser({balance: 2});
+    userToInvite = await generateUser();
+  });
+
+  context('leaving guilds', () => {
+    let guild;
+
+    beforeEach(async () => {
+      guild = await user.post('/groups', {
+        name: 'Test Public Guild',
+        type: 'guild',
+      });
+      await user.post(`/groups/${guild._id}/invite`, {
+        uuids: [userToInvite._id],
+      });
+    });
+
+    it('lets user leave a guild', async () => {
+      await userToInvite.post(`/groups/${guild._id}/join`);
+      await userToInvite.post(`/groups/${guild._id}/leave`);
+
+      let userThatLeftGuild = await userToInvite.get('/user');
+
+      expect(userThatLeftGuild.guilds).to.be.empty;
+    });
+
+    it('sets a new group leader when leader leaves a guild', async () => {
+      await userToInvite.post(`/groups/${guild._id}/join`);
+      await user.post(`/groups/${guild._id}/leave`);
+
+      let guildWithNewLeader = await userToInvite.get(`/groups/${guild._id}`);
+
+      expect(guildWithNewLeader.leader._id).to.equal(userToInvite._id);
+    });
+
+    it('removes a group and invitations when the last member leaves', async () => {
+      await user.post(`/groups/${guild._id}/leave`);
+
+      let groups = await user.get('/groups?type=party,privateGuilds,publicGuilds,tavern');
+      let userWithoutInvitation = await userToInvite.get('/user');
+
+      expect(_.findIndex(groups, {_id: guild._id})).to.equal(-1);
+      expect(userWithoutInvitation.invitations.guilds).to.be.empty;
```

Comment:
Oh, it's not a public guild, but the guild name is 'test public guild' so I was confused.